discovery/dns/dns.go

```diff
@@ -317,19 +317,26 @@ func askServerForName(name string, queryType uint16, client *dns.Client, servAdd
 	}
 
 	response, _, err := client.Exchange(msg, servAddr)
-	if err == dns.ErrTruncated {
+	if err != nil {
+		if err == dns.ErrId {
+			if response != nil { // response can be nil here, just be paranoid
```

Comment:
This check is of little consequence since we return `nil` anyway. I think we are better off without this extra if check.

Reply:
From the prometheus SD perspective.

Reply:
oh the conditional is reversed, oops.

Reply:
there, the conditional is fixed. as mentioned prior, if we remove this and the response is nil, then the `response.Id` dereference will panic the running goroutine.